### DISCUSSION TITLE
RoderickShao_notebook/%E8%AE%A1%E7%AE%97%E6%9C%BA%E5%9F%BA%E7%A1%80%E8%AF%BE/%E9%AB%98%E7%BA%A7%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84%E4%B8%8E%E7%AE%97%E6%B3%95%E5%88%86%E6%9E%90/Lecture%2015/

### DISCUSSION THREAD

**Discussion by giscus** [RoderickShao/RoderickShao_notebook#17] [Announcements]
# RoderickShao_notebook/%E8%AE%A1%E7%AE%97%E6%9C%BA%E5%9F%BA%E7%A1%80%E8%AF%BE/%E9%AB%98%E7%BA%A7%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84%E4%B8%8E%E7%AE%97%E6%B3%95%E5%88%86%E6%9E%90/Lecture%2015/

RoderickShao's NoteBook

https://roderickshao.github.io/RoderickShao_notebook/%E8%AE%A1%E7%AE%97%E6%9C%BA%E5%9F%BA%E7%A1%80%E8%AF%BE/%E9%AB%98%E7%BA%A7%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84%E4%B8%8E%E7%AE%97%E6%B3%95%E5%88%86%E6%9E%90/Lecture%2015/

**Comment by Provence-24** [comment]
写的非常认真，也非常棒！完结撒花
尝试梳理：
1.减少pass：增加k（一次pass表示一组磁头的移动或称定位）、更长的run（详见4）
2.减少type：合理利用空间 斐波那契
3.CPU比较和IO并行操作：2k个input，2个output，一半存现在，一半预留为buffer
4.更长的run：用堆实现，堆中元素全红或没有剩余元素时，新开一个run
深入思考：
1.run与block的区别 前者有序，前者是后者经过第一次pass的结果
2.k-way并且未进行优化时type的数量：2k （原因：均匀取时为最坏情况，因为没有一个input空出来，所以都得用output的type，因此input和output各k个，共2k）
3.近似斐波那契那里我感觉应该是补充元素，使其成为最近的斐波那契数。
